scripts/biomedical/humanMetabolomeDatabase/format_hmdb_protein.py

```diff
@@ -0,0 +1,495 @@
+    Returns:
+        df_remained: dataframe with protein dcid missing
+    """
+    all_results = []
+    attribute = list(set(df_input[attribute_column].unique()) - set([np.nan]))
+    for alist in split_list(attribute, QUERY_SIZE):
+        attribute = np.array(alist)
+        query_str = """
+        SELECT DISTINCT ?protein
+        WHERE {{
+        ?protein typeOf Protein .
+        ?protein {0} {1} .
+        }}
+        """.format(protein_attribute, attribute)
+        result = dc.query(query_str)
+        all_results.extend(result)
+    protein_dcid = np.array([res["?protein"] for res in all_results])
+    df_property = pd.DataFrame(columns=[attribute_column, "protein"])
+    df_property["protein"] = protein_dcid
+
+    property_dict = dc.get_property_values(protein_dcid,
+                                           prop=protein_attribute,
+                                           out=True)
+    df_property[attribute_column] = df_property["protein"].map(property_dict)
+    df_property = df_property.explode(attribute_column)
+    property_dict = df_property.set_index(attribute_column).to_dict()["protein"]
+    df_original[protein_dcid_column] = df_original[protein_dcid_column].\
+                                fillna(df_original[attribute_column].map(property_dict))
+    df_remained = df_original[df_original[protein_dcid_column].isna()]
+    return df_remained
+
+
+def format_comparment(compartment: str) -> str:
+    """format subcellular_location naming
+    to datacommons format: drop bad values,
+    synchronize synonyms, Enum format
+    Args:
+        compartment: compartment name
+    Returns:
+        compartment: formatted compartment
+    """
+    modify_dict = {"Cytoplasmic": "Cytoplasm", "Cytoplasmic side": "Cytoplasm"}
+    drop_list = [
+        "Processed beta-1", "Transferrin receptor protein 1",
+        "Leucyl-cystinyl aminopeptidase", " Alpha-1",
+        "Calcium-activated chloride channel regulator 2",
+        "Tumor necrosis factor", "T-cell surface glycoprotein CD1e"
+    ]
+    if compartment in drop_list:
+        return np.nan
+    elif compartment in modify_dict.keys():
+        return "dcs:CellularCompartment" + modify_dict[compartment]
+    else:
+        compartment = compartment.split(":")[-1]
+        compartment = compartment.split("(")[0]
+        compartment = compartment.title()
+        compartment = compartment.replace(" ", "")
+        compartment = compartment.replace("-", "")
+        compartment = "dcs:CellularCompartment" + compartment
+        return compartment
+
+
+def convert_float_to_int(df_input: pd.DataFrame, column: str) -> None:
+    """convert type of pd.Datafarme column
+    from float to int, ignore missing values
+    Args:
+        df: pandas.DataFrame
+        column: column name
+    Returns:
+        None
+    """
+    df_input[column] = df_input[column].fillna(-1)
+    df_input[column] = df_input[column].astype(int)
+    df_input[column] = df_input[column].astype(str)

```

Comment:
is there a reason for converting the column to ints first instead of just converting to str right away?

Reply:
@khoahoang1891999